content/docs/forms.md

```diff
@@ -254,31 +254,31 @@ class Reservation extends React.Component {
 }
 ```
 
-[**Try it on CodePen**](https://codepen.io/gaearon/pen/wgedvV?editors=0010)
+[**इसे CodePen पर आज़माएँ**](https://codepen.io/gaearon/pen/wgedvV?editors=0010)
 
-Note how we used the ES6 [computed property name](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names) syntax to update the state key corresponding to the given input name:
+ध्यान दीजिये की हमने दिए गए इनपुट नाम के लिए state key को अपडेट करने के लिए ES6 [कंप्यूटेड प्रॉपर्टी नाम](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names) सिंटेक्स का उपयोग किया है।
 
 ```js{2}
 this.setState({
   [name]: value
 });
 ```
 
-It is equivalent to this ES5 code:
+यह निचे दिए गए ES5 code के समान है :
 
 ```js{2}
 var partialState = {};
 partialState[name] = value;
 this.setState(partialState);
 ```
 
-Also, since `setState()` automatically [merges a partial state into the current state](/docs/state-and-lifecycle.html#state-updates-are-merged), we only needed to call it with the changed parts.
+और क्यूंकि `setState()` अपनेआप [एक आंशिक state को अभी के state में मिला देता है](/docs/state-and-lifecycle.html#state-updates-are-merged)], इसीलिए हमे इसे सिर्फ बदले हुए हिस्सों के लिए उपयोग करना होगा।
 
-## Controlled Input Null Value {#controlled-input-null-value}
+## कंट्रोल्ड इनपुट Null वैल्यू {#controlled-input-null-value}
 
-Specifying the value prop on a [controlled component](/docs/forms.html#controlled-components) prevents the user from changing the input unless you desire so. If you've specified a `value` but the input is still editable, you may have accidentally set `value` to `undefined` or `null`.
+एक [कंट्रोल्ड कौम्पोनॅन्ट](/docs/forms.html#controlled-components) पर वैल्यू प्रॉपर्टी लगा देने से हम यूजर को इनपुट बदलने से तब तक रोक सकते हैं जब तक हम ना चाहें। अगर हमने `value` प्रॉपर्टी दी है लेकिन फिर भी इनपुट को बदला जा सकता है, तो इसका मतलब यह है की हमने गलती से `value` को या तो `undefined` या `null` सेट कर दिया है।  
 
-The following code demonstrates this. (The input is locked at first but becomes editable after a short delay.)
+निचे दिया गया कोड यही दर्शाता है। (पहले इनपुट को बदला नहीं जा सकता था लेकिन थोड़ी देर के बाद उसे बदला जा सकता है)
```

Comment:
`The input is locked at first but becomes editable after a short delay.` => `इनपुट पहले लॉक रहता है लेकिन थोड़े डिले के बाद एडिटेब्ल हो जाता है।`